src/components/kytos/misc/StatusMenu.vue

```diff
@@ -0,0 +1,764 @@
+        return "selectedButton"
+      }
+    },
+    show_infoPanel(s) {
+      /**
+       * @event showInfoPanel
+       * @param {Object} s - The switch data from the current row used to generate the info panel.
+       * @description Displays info panel when table row is clicked.
+       */
+      var content = {component: 'kytos-topology-k-info-panel-switch_info',
+                     content: this.switches[s],
+                     icon: "cog",
+                     title: "Switch Details",
+                     subtitle: this.switches[s].connection}
+      this.$kytos.$emit("showInfoPanel", content)
+    },
+    /**
+     * @param {string} data - The name of the selected data.
+     * @description Changes the current data value stored in memory to the data param. 
+     *              Split for each individual table: Switches, Links, and Interfaces.
+     */
+    changeSwitchData(data) {
+      this.currentSwitchData = data
+    },
+    changeLinkData(data) {
+      this.currentLinkData = data
+    },
+    changeInterfaceData(data) {
+      this.currentInterfaceData = data
+    },
+    /**
+     * @param {Object} item - The normal table row object containing:
+     * @param {string} item.dpid - The data path id of the device the Object references.
+     * @param {string} item.id - The id of the device the Object references.
+     * @param {string} index - The index of the current table row Object. Usually id or name.
+     * @description Adds and Removes elements from the respective "selected" list based on id.
+     *              Split for each individual table: Switches, Links, and Interfaces.
+     */
+    add_remove_Switches(item, index) {
+      if (this.selectedSwitches[index]?.dpid == item.dpid) {
+        delete this.selectedSwitches[index]
+      } else {
+        this.$set(this.selectedSwitches, index, item);
+      }
+      this.change = this.change * -1
+      this.$forceUpdate();
+    },
+    add_remove_Links(item, index) {
+      if (this.selectedLinks[index]?.id == item.id) {
+        delete this.selectedLinks[index]
+      } else {
+        this.$set(this.selectedLinks, index, item);
+      }
+      this.change = this.change * -1
+      this.$forceUpdate();
+    },
+    add_remove_Interfaces(item, index) {
+      if (this.selectedInterfaces[index]?.id == item.id) {
+        delete this.selectedInterfaces[index]
+      } else {
+        this.$set(this.selectedInterfaces, index, item);
+      }
+      this.change = this.change * -1
+      this.$forceUpdate();
+    },
+    //API Requests for Switch, Link, and Interface data.
+    get_topology() {
+      var self = this
+      let request = $.ajax({
+                       type:"GET",
+                       dataType: "json",
+                       url: this.$kytos_server_api + "kytos/topology/v3",
+                       async: true});
+      request.done(function(data) {
+        self.switches = data['topology']['switches']
+        self.links = data['topology']['links']
```

Comment:
Heads-up: the interface_id, did and content_switch properties on the interface were added because they will be needed for my other comment on the click Event to open the InfoPanel to view/edit the interface details.